### DISCUSSION TITLE
[HxSearchBox] - Failed to fetch dynamically imported module - wrong url to HxSearchBox.js

### DISCUSSION THREAD

**Discussion by ghost** [havit/Havit.Blazor#411] [Q&A] [answered]
Hi,

I have a small problem with the HxSearchBox.

Can see in the console, that a bad link is generates:

`Microsoft.JSInterop.JSException: Failed to fetch dynamically imported module: https://localhost:7203/CreateApplication_content/Havit.Blazor.Components.Web.Bootstrap/HxSearchBox.js?v=2.0.0%2B284c63439580bd101aa55baacebc09c1b4cbd7dc`

The problem is that the url generates is :

'https://localhost:7203/CreateApplication_content/Havit.Blazor.Components.Web.Bootstrap/HxSearchBox.js?v=2.0.0%2B284c63439580bd101aa55baacebc09c1b4cbd7dc'

and not:

'https://localhost:7203/_content/Havit.Blazor.Components.Web.Bootstrap/HxSearchBox.js?v=2.0.0%2B284c63439580bd101aa55baacebc09c1b4cbd7dc'

Is it a bad configuration in my end?

Best,
CU

**Comment by hakenr** [comment]
Hello,
do you have `<base href="/CreateApplication/" />` element in your application?
Can you please provide us with a sample code reproducing the error?

**Comment by hakenr** [comment]
Also verify you have the `<StaticWebAssetBasePath>CreateApplication</StaticWebAssetBasePath>` in your `.csproj` file.

**Comment by ghost** [comment]
Ohh actually this is a nested use, so the CprInfoEditor component includes the HxSearchBox  (actually just the sample code from documentation).

```
@page "/CreateApplication"

<div class="row">
    <div class="col-12">
        <CprInfoEditor/>
    </div>
</div>
```

So maybe that's the reason. Is there a work around for that use case :-)

/CU

**Comment by hakenr** [comment, marked as answer]
Nesting of components should not bring such issues.
For Blazor application to work properly, you have to have the `<base ... />` element somewhere in you page.
It usually sits in `index.html`, `_host.cshtml` or `_layout.cshtml` (depending on your setup) and is responsible for correct URL resolution on client-side.

**Comment by ghost** [comment]
Thank you for pointing me in the right direction :-)

The problem was that my _Host.cshtml did not contain the` <base href="" />` tag at all. 
Maybe the .NET 6 template does not add that by default?

In another solution I could see `<base href="~/"/>` and after adding that to _Host.cshtm it all worked very well.
